hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlPartitionPruningE2ETest.java

```diff
@@ -0,0 +1,419 @@
+import com.hazelcast.sql.SqlRow;
+import com.hazelcast.sql.SqlStatement;
+import com.hazelcast.sql.impl.QueryId;
+import com.hazelcast.sql.impl.SqlServiceImpl;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContextImpl;
+import com.hazelcast.sql.impl.optimizer.SqlPlan;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class SqlPartitionPruningE2ETest extends SqlTestSupport {
+    private static ExpressionEvalContext EEC;
+
+    private SqlServiceImpl sqlService;
+    private PlanExecutor planExecutor;
+    private String mapName;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        initialize(3, null);
+        EEC = new ExpressionEvalContextImpl(
+                emptyList(),
+                Util.getSerializationService(instance()),
+                Util.getNodeEngine(instance()));
+    }
+
+    @Before
+    public void before() throws Exception {
+        sqlService = (SqlServiceImpl) instance().getSql();
+        planExecutor = sqlService.getOptimizer().getPlanExecutor();
+        mapName = randomName();
+    }
+
+    @Test
+    public void when_scanWithSimplePruningKey_then_prunable() {
+        final String query = "SELECT * FROM " + mapName + " WHERE f0 = 2";
+
+        instance().getConfig().addMapConfig(
+                new MapConfig(mapName).setPartitioningAttributeConfigs(List.of(
+                        new PartitioningAttributeConfig("f0")
+                )));
+        IMap<Pojo, String> map = instance().getMap(mapName);
+        createMapping(mapName, Pojo.class, String.class);
+
+        map.put(new Pojo(2, 2, 2), "2");
+
+        SqlStatement sql = new SqlStatement(query);
+        SqlPlan plan = sqlService.prepare(
+                sql.getSchema(),
+                query,
+                sql.getParameters(),
+                SqlExpectedResultType.ROWS
+        );
+
+        assertInstanceOf(SqlPlanImpl.SelectPlan.class, plan);
+        SqlPlanImpl.SelectPlan selectPlan = (SqlPlanImpl.SelectPlan) plan;
+
+        var optResult = planExecutor.tryUsePrunability(selectPlan, EEC);
```

Comment:
this definitely tests more that before (it is possible now to write assertions on the DAG), but it still does not test all assumptions on the Jet-SQL interface, in particular correctness of the `JobConfig` (we might create it in a wrong way, forget to invoke tryUsePrunability in production code etc.)